### PR TITLE
Using grunt for magento tasks

### PR DESCRIPTION
**Pull request by atwixfirster**
## Scope of this PR

- __Technical__: Changes to technical content/code/processes/naming conventions (any change to technical content)

## Purpose of this PR

PR adds additional explanations about content of `grunt-config.json` file. 
PR adds information how to avoid `Cannot find module /<PATH_TO_MAGENTO2_STORE>/dev/tools/grunt/configs'` error 

error

within grunt compilation with default `grunt-config.json` settings. 

## Affected URLs

- https://devdocs.magento.com/guides/v2.2/frontend-dev-guide/tools/using_grunt.html
- https://devdocs.magento.com/guides/v2.3/frontend-dev-guide/tools/using_grunt.html

Thank you!

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by atwixfirster** [comment]
> Hi @atwixfirster. By default, with a correct gruntfile this step is not required. Could you doublecheck this point, please?
> Thank you

Hey, @rogyar !

I've verified this issue on clear M2.

**Preconditions:**
Rename the following files in your Magento root directory:
- `package.json.sample` to `package.json`
- `Gruntfile.js.sample` to `Gruntfile.js`
- `grunt-config.json.sample` to `grunt-config.json`

**Steps to reproduce:**
**1.** `curl -sL https://deb.nodesource.com/setup_8.x | sudo bash -`
**2.** `sudo apt-get install gcc g++ make`
**3.** `sudo apt-get install -y nodejs`
**4.** `sudo npm install -g grunt-cli`
**5.** `npm install`
**6.** `npm update`
**7.**`grunt clean`

**Actual result:**

**Comment by hostep** [comment]
@atwixfirster: if I remember correctly, you also need to copy/rename the file `dev/tools/grunt/configs/themes.js` to `dev/tools/grunt/configs/local-themes.js`

I prefer to not rename files but copy them, just to keep the originals around.

See this other open issue for more details (I don't know how accurate this still is though): https://github.com/magento/devdocs/issues/1537

**Comment by atwixfirster** [comment]
> @atwixfirster: if I remember correctly, you also need to copy/rename the file `dev/tools/grunt/configs/themes.js` to `dev/tools/grunt/configs/local-themes.js`
> 
> I prefer to not rename files but copy them, just to keep the originals around.
> 
> See this other open issue for more details (I don't know how accurate this still is though): #1537

Thank you for notice, @hostep !

I've made changes in this PR. Could you please review?

Thank you!

**Comment by hostep** [comment]
@atwixfirster: I would simply add the extra file to step 3, and maybe change the word `Rename` to `Copy`.
And then in the next part labelled `Grunt configuration file`, we should change `themes.js` to `local-themes.js`.

I would then also completely get rid of everything starting from `The problem with using the default configuration files...` and everything (the entire rest of the page) underneath it. This whole page is very confusing, default grunt config vs custom grunt config ... Everybody needs the custom grunt config in my opinion.

But that's just my opinion, others are welcome too of course :)

**Comment by atwixfirster** [comment]
> @atwixfirster: I would simply add the extra file to step 3, and maybe change the word `Rename` to `Copy`.
> And then in the next part labelled `Grunt configuration file`, we should change `themes.js` to `local-themes.js`.
> 
> I would then also completely get rid of everything starting from `The problem with using the default configuration files...` and everything (the entire rest of the page) underneath it. This whole page is very confusing, default grunt config vs custom grunt config ... Everybody needs the custom grunt config in my opinion.
> 
> But that's just my opinion, others are welcome too of course :)

Hey, @hostep !

All your suggestion have been implemented! 

@rogyar , could you please verify and let us know your thoughts.

Thank you!

File: guides/v2.2/frontend-dev-guide/tools/using_grunt.md [hunk 1]
```diff
@@ -26,11 +26,11 @@ Magento has built-in Grunt tasks configured, but there are still several steps y
 
        npm install -g grunt-cli
 
-3. Rename the following files in your Magento root directory:
+3. Copy the following files in your Magento root directory:
```
[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/frontend-dev-guide/tools/using_grunt.md]: I meant copy file to the same directory but  with a new name 😊

[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/frontend-dev-guide/tools/using_grunt.md]: > If you are simple copying the files to the new location, then simply list the files below.

I've made changes regarding your notice. Could you please verify?

Thank you, @dobooth !

File: guides/v2.2/frontend-dev-guide/tools/using_grunt.md [hunk 2]
```diff
@@ -41,7 +41,9 @@ Magento has built-in Grunt tasks configured, but there are still several steps y
 
 ## Grunt configuration file {#grunt_config}
 
-If installed as described above, Grunt will use the default configuration files located in the `dev/tools/grunt/configs/` directory. You can define your theme in the `themes.js` file located in the above directory. The following shows an example of how you can define your theme.
+Copy `themes.js` to `local-themes.js` in the `dev/tools/grunt/configs/` directory.
```
[the remark this answers is not in the document]
Comment (atwixfirster) [reply guides/v2.2/frontend-dev-guide/tools/using_grunt.md]: > Does this mean "Copy the contents of themes.js into local-themes.js?

**fixed**

Please verify.

Thank you!

**Pull request merged**